test/e2e/utils_test.go

```diff
@@ -764,3 +764,31 @@ func pollTillDeploymentAvailable(ctx context.Context, clientSet *kubernetes.Clie
 
 	return err
 }
+
+// VerifyContainerResources verifies that all containers in a pod have resources matching the expected configuration
+func VerifyContainerResources(pod corev1.Pod, expectedResources corev1.ResourceRequirements) error {
+	if len(pod.Spec.Containers) == 0 {
+		return fmt.Errorf("no containers found in pod %s", pod.Name)
+	}
+
+	for _, container := range pod.Spec.Containers {
```

Comment:
nit: we don't have multiple containers as of today, maybe not in future too. But would it better to verify these on specific container using container name?